recipes/nasm/all/test_package/conanfile.py

```diff
@@ -1,18 +1,26 @@
+from conan import ConanFile
+from conan.tools.build import can_run
 import os
-from conans import ConanFile, tools
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "arch",
+    # settings = "os", "arch",
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
```

Comment:
```suggestion
    def build_requirements(self):
        self.tool_requires(self.tested_reference_str)
```

nasm will be used as a build requirement

Reply:
oooh